File: app/src/main/java/com/sharpdroid/registroelettronico/BottomSheet/OrderMedieBS.java
package com.sharpdroid.registroelettronico.BottomSheet;

import android.content.Context;
import android.os.Bundle;
import android.os.Handler;
import android.os.Looper;
import android.support.annotation.Nullable;
import android.support.design.widget.BottomSheetDialogFragment;
import android.support.v4.app.Fragment;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.sharpdroid.registroelettronico.R;

/**
 * <p>A fragment that shows a list of items as a modal bottom sheet.</p>
 * <p>You can show this modal bottom sheet from your activity like this:</p>
 * <pre>
 *     OrderMedieBS.newInstance(30).show(getSupportFragmentManager(), "dialog");
 * </pre>
 * <p>You activity (or fragment) needs to implement {@link OrderListener}.</p>
 */
public class OrderMedieBS extends BottomSheetDialogFragment {
    private OrderListener mListener;

    private final int[] texts = {
            R.string.nome_bs,
            R.string.media_dec_bs,
            R.string.media_cre_bs
    };

    private final int[] images = {
            R.drawable.ic_title,
            R.drawable.ic_timeline,
            R.drawable.ic_timeline
    };

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.bottom_sheet_list_order_medie, container, false);
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        final RecyclerView recyclerView = (RecyclerView) view.findViewById(R.id.list);
        recyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
        recyclerView.setAdapter(new ItemAdapter(3));
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        final Fragment parent = getParentFragment();
        if (parent != null) {
            mListener = (OrderListener) parent;
        } else {
            mListener = (OrderListener) context;
        }
    }

    @Override
    public void onDetach() {
        mListener = null;
        super.onDetach();
    }

    public interface OrderListener {
        void onItemClicked(int position);
    }

    private class ViewHolder extends RecyclerView.ViewHolder {

        final TextView text;
        final ImageView image;

        ViewHolder(LayoutInflater inflater, ViewGroup parent) {
            super(inflater.inflate(R.layout.bottom_sheet_order_medie, parent, false));
            text = (TextView) itemView.findViewById(R.id.text);
            image = (ImageView) itemView.findViewById(R.id.image);
            itemView.setOnClickListener(v -> {
                if (mListener != null) {
                    new Handler(Looper.getMainLooper()).postDelayed(() -> {
                        mListener.onItemClicked(getAdapterPosition());
                        dismiss();
                    }, 400);
                }
            });
        }
    }

    private class ItemAdapter extends RecyclerView.Adapter<ViewHolder> {

        private final int mItemCount;

        ItemAdapter(int itemCount) {
            mItemCount = itemCount;
        }

        @Override
        public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
            return new ViewHolder(LayoutInflater.from(parent.getContext()), parent);
        }

        @Override
        public void onBindViewHolder(ViewHolder holder, int position) {
            holder.text.setText(getString(texts[position]));
            holder.image.setImageResource(images[position]);
        }

        @Override
        public int getItemCount() {
            return mItemCount;
        }

    }

}

File: app/src/main/java/com/sharpdroid/registroelettronico/BottomSheet/OrderMedieBS.kt
package com.sharpdroid.registroelettronico.BottomSheet

import android.content.Context
import android.os.Bundle
import android.support.design.widget.BottomSheetDialogFragment
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.sharpdroid.registroelettronico.BottomSheet.OrderMedieBS.OrderListener
import com.sharpdroid.registroelettronico.R
import kotlinx.android.synthetic.main.bottom_sheet_order_medie.view.*

/**
 *
 * A fragment that shows a list of items as a modal bottom sheet.
 *
 * You can show this modal bottom sheet from your activity like this:
 * <pre>
 * OrderMedieBS.newInstance(30).show(getSupportFragmentManager(), "dialog");
</pre> *
 *
 * You activity (or fragment) needs to implement [OrderListener].
 */
class OrderMedieBS : BottomSheetDialogFragment() {
    private val texts = intArrayOf(R.string.nome_bs, R.string.media_bs, R.string.nvoti_bs)
    private val images = intArrayOf(R.drawable.ic_title, R.drawable.ic_timeline, R.drawable.ic_timeline)
    private var mListener: OrderListener? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.bottom_sheet_list_order_medie, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        val recyclerView = view.findViewById<RecyclerView>(R.id.list)
        recyclerView?.layoutManager = LinearLayoutManager(context)
        recyclerView?.adapter = ItemAdapter(3, mListener!!)
    }

    override fun onAttach(context: Context?) {
        super.onAttach(context)
        val parent = parentFragment
        mListener = if (parent != null) {
            parent as OrderListener
        } else {
            context as OrderListener?
        }
    }

    override fun onDetach() {
        mListener = null
        super.onDetach()
    }

    interface OrderListener {
        fun onItemClicked(position: Int)
    }


    private inner class ItemAdapter internal constructor(private val mItemCount: Int, val clickListener: OrderListener) : RecyclerView.Adapter<ItemAdapter.ViewHolder>() {
        private inner class ViewHolder(inflater: LayoutInflater, parent: ViewGroup) : RecyclerView.ViewHolder(inflater.inflate(R.layout.bottom_sheet_order_medie, parent, false)) {
            fun bind(text: String, image: Int) {
                itemView.text.text = text
                itemView.image.setImageResource(image)
                itemView.setOnClickListener {
                    itemView.postDelayed({
                        clickListener.onItemClicked(adapterPosition)
                        dismiss()
                    }, 200)
                }
            }
        }

        override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
            return ViewHolder(LayoutInflater.from(parent.context), parent)
        }

        override fun onBindViewHolder(holder: ViewHolder, position: Int) {
            holder.bind(getString(texts[position]), images[position])
        }

        override fun getItemCount(): Int {
            return mItemCount
        }
    }
}
